x/concentrated-liquidity/position.go

```diff
@@ -401,3 +439,49 @@ func (k Keeper) validatePositionsAndGetTotalLiquidity(ctx sdk.Context, owner sdk
 	}
 	return basePosition.PoolId, basePosition.LowerTick, basePosition.UpperTick, totalLiquidity, nil
 }
+
+// GetConcentratedLockupDenom returns the concentrated lockup denom for a given pool and position.
+func GetConcentratedLockupDenom(poolId, positionId uint64) string {
+	return fmt.Sprintf("cl/pool/%d/%d", poolId, positionId)
+}
+
+// GetPositionIdToLock returns the positionId to lock mapping in state.
+func (k Keeper) GetPositionIdToLock(ctx sdk.Context, positionId uint64) (uint64, error) {
+	store := ctx.KVStore(k.storeKey)
+
+	// Get the position ID to key mapping.
+	key := types.KeyPositionIdForLock(positionId)
+	value := store.Get(key)
+	if value == nil {
+		return 0, types.PositionIdToLockNotFoundError{PositionId: positionId}
+	}
+
+	return sdk.BigEndianToUint64(value), nil
+}
+
+// SetPositionIdToLock sets the positionId to lock mapping in state.
+func (k Keeper) SetPositionIdToLock(ctx sdk.Context, positionId, underlyingLockId uint64) {
+	store := ctx.KVStore(k.storeKey)
+
+	// Get the position ID to key mapping.
+	key := types.KeyPositionIdForLock(positionId)
+	store.Set(key, sdk.Uint64ToBigEndian(underlyingLockId))
+}
+
+// RemovePositionIdToLock removes the positionId to lock mapping from state.
+func (k Keeper) RemovePositionIdToLock(ctx sdk.Context, positionId uint64) {
+	store := ctx.KVStore(k.storeKey)
+
+	// Get the position ID to key mapping.
+	key := types.KeyPositionIdForLock(positionId)
+
+	// Delete the mapping from state.
+	store.Delete(key)
+}
+
+// positionHasUnderlyingLockInState checks if a given positionId has a corresponding lock in state.
+func (k Keeper) positionHasUnderlyingLockInState(ctx sdk.Context, positionId uint64) bool {
+	// Get the lock ID for the position.
+	_, err := k.GetPositionIdToLock(ctx, positionId)
+	return !errors.Is(err, types.PositionIdToLockNotFoundError{PositionId: positionId})
+}
```

Comment:
WDYT about?

Slightly worried that we get false positive when some random error occurs

```suggestion
func (k Keeper) positionHasUnderlyingLockInState(ctx sdk.Context, positionId uint64) (bool, error) {
	// Get the lock ID for the position.
	_, err := k.GetPositionIdToLock(ctx, positionId)
	if err == nil || !errors.Is(err, types.PositionIdToLockNotFoundError{PositionId: positionId}) {
		return true, nil
	}
	return false, err
}
```

Reply:
added error here https://github.com/osmosis-labs/osmosis/pull/4882/commits/d0f00c83f04c2d6fd37a60de13567bcca7841c61